quickwit-core/src/indexing/statistics.rs

```diff
@@ -91,6 +99,8 @@ impl StatisticsCollector {
             num_published_splits: 0,
             total_bytes_processed: 0,
             total_size_splits: 0,
+            start_time: Instant::now(),
+            stdout: stdout(),
```

Comment:
I'd rather have, an Arc<StatisticsCollector> that contains the model of the data.

It contains start_time, and all of the counters as Atomics. We can use super relaxed ordering (I guess... let's just make we don't divide anything by 0 as a result!?).

We could then have a task dedicating to display these stats. That task would be spawned from the cli function.

Reply:
I thought of atomic counters but i was thinking keeping this way is simpler. I only used channel and events here because of the requirement of displaying a kind of live progress on statistics. It was a way to keep all this in on place.